README.md

```diff
@@ -123,5 +123,146 @@ LabelCell(key: "company", state: LabelState(text: "Shopify",
+
+```swift
+
+class MessageView: UIView {
+
+    let profilePicture = UIImageView()
+    let nameLabel = UILabel()
+    let messageLabel = UILabel()
+    let timeStampLabel = UILabel()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        backgroundColor = .white
+        setupConstraints()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func setupConstraints() {
+        profilePicture.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(profilePicture)
+
+        addSubview(nameLabel)
+        addSubview(messageLabel)
+
+        timeStampLabel.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(timeStampLabel)
+
+        let stackView = UIStackView(arrangedSubviews: [nameLabel, messageLabel])
+        stackView.axis = .vertical
+        stackView.alignment = .fill
+        stackView.spacing = 3
+        stackView.distribution = .fill
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(stackView)
+
+        let constraints: [NSLayoutConstraint] = [
+        profilePicture.heightAnchor.constraint(equalToConstant: 60),
+        profilePicture.widthAnchor.constraint(equalToConstant: 60),
+        profilePicture.leadingAnchor.constraint(equalTo: leadingAnchor),
+        profilePicture.topAnchor.constraint(equalTo: topAnchor, constant: 12),
+        profilePicture.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -12),
+        stackView.centerYAnchor.constraint(equalTo: profilePicture.centerYAnchor),
+        stackView.topAnchor.constraint(greaterThanOrEqualTo: topAnchor),
+        stackView.bottomAnchor.constraint(lessThanOrEqualTo: bottomAnchor),
+        stackView.leadingAnchor.constraint(equalTo: profilePicture.trailingAnchor, constant: 12),
+        stackView.trailingAnchor.constraint(lessThanOrEqualTo: timeStampLabel.leadingAnchor, constant: -12),
+        timeStampLabel.trailingAnchor.constraint(equalTo: trailingAnchor),
+        timeStampLabel.topAnchor.constraint(equalTo: topAnchor)
+        ]
+        NSLayoutConstraint.activate(constraints)
+
+        profilePicture.layer.borderWidth = 1
+        profilePicture.layer.borderColor = UIColor.black.cgColor
+        profilePicture.layer.cornerRadius = 30
+        profilePicture.backgroundColor = UIColor.gray
+
+        nameLabel.font = UIFont.systemFont(ofSize: 18, weight: .medium)
+        messageLabel.font = UIFont.systemFont(ofSize: 15, weight: .light)
+
+        timeStampLabel.font = UIFont.systemFont(ofSize: 12, weight: .light)
+        timeStampLabel.setContentHuggingPriority(.required, for: .horizontal)
+    }
+}
+```
+
+Next, we'll build the `MessageViewState` struct that will take care of setting all the views properties:
+
+```swift
+
+struct Message {
+    let name: String
+    let image: UIImage
+    let date: Date
+    let text: String
+}
+
+struct MessageViewState: Equatable {
+
```

Comment:
```suggestion
```

(Same for all the places that there is a line between the declaration and the code) :)